### PR TITLE
fix: fixed autonaming item without default company

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
fixed Company error while creating item if **Default Company** not being set in **Global Defaults**

Ref: [Asana](https://app.asana.com/0/1192407896547052/1198958632787845)

### CODE REVIEW

File: bloomstack_core/hook_events/item.py [hunk 1]
```diff
@@ -28,14 +28,21 @@ def autoname(item, method=None):
...
 	# Get abbreviations
-	company_abbr = get_company_default(get_default_company(), "abbr")
-	brand_abbr = get_abbr(item.brand, max_length=len(company_abbr))
-	brand_abbr = brand_abbr if company_abbr != brand_abbr else None
+	default_company = get_default_company()
+	if default_company:
+		company_abbr = get_company_default(default_company, "abbr")
+		brand_abbr = get_abbr(item.brand, max_length=len(company_abbr))
+		brand_abbr = brand_abbr if company_abbr != brand_abbr else None
+	brand_abbr = get_abbr(item.brand)
```
Comment (neilLasrado) [inline bloomstack_core/hook_events/item.py]: This will override the brand_abbr set on L34 & L35

**Review by neilLasrado — CHANGES_REQUESTED** [review]

**Comment by MyuddinKhatri** [comment]
@neilLasrado can you please have a look?

File: bloomstack_core/hook_events/item.py [hunk 2]
```diff
@@ -28,14 +28,22 @@ def autoname(item, method=None):
...
-	company_abbr = get_company_default(get_default_company(), "abbr")
-	brand_abbr = get_abbr(item.brand, max_length=len(company_abbr))
-	brand_abbr = brand_abbr if company_abbr != brand_abbr else None
+	default_company = get_default_company()
+	if default_company:
+		company_abbr = get_company_default(default_company, "abbr")
+		brand_abbr = get_abbr(item.brand, max_length=len(company_abbr))
+		brand_abbr = brand_abbr if company_abbr != brand_abbr else None
+	if not default_company:
+		brand_abbr = get_abbr(item.brand)
```
Comment (sahil28297) [inline bloomstack_core/hook_events/item.py]: use else instead of `if not default company:`

File: bloomstack_core/hook_events/item.py [hunk 3]
```diff
@@ -28,14 +28,22 @@ def autoname(item, method=None):
...
 	item_name_abbr = get_abbr(item.item_name, 3)
 
-	params = list(filter(None, [company_abbr, brand_abbr, item_group_abbr, item_name_abbr]))
-	item_code = "-".join(params)
+	if default_company:
+		params = list(filter(None, [company_abbr, brand_abbr, item_group_abbr, item_name_abbr]))
+		item_code = "-".join(params)
+	else:
+		params = list(filter(None, [brand_abbr, item_group_abbr, item_name_abbr]))
+		item_code = "-".join(params)
```
Comment (sahil28297) [inline bloomstack_core/hook_events/item.py]: move lines 38 and 39 above line 30 and combine the if statement here with the above one

**Review by sahil28297 — CHANGES_REQUESTED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**